.github/workflows/create-slot-name.yml

```diff
@@ -0,0 +1,58 @@
+name: Get slot name from branch
+on:
+  workflow_call:
+    inputs:
+      webApp:
+        description: "Web App Name"
+        required: true
+        type: string
+      branch-name:
+        required: false
+        type: string
+        description: 'The name of the branch to process. If not provided, it will be automatically determined.'
+    outputs:
+      slot-name:
+        value: ${{ jobs.create-deployment-slot-name.outputs.slot-name }}
+jobs:
+  create-deployment-slot-name:
+    name: Create deployment slot name
+    runs-on: ubuntu-latest
+    outputs:
+      slot-name: ${{ steps.truncate-branch-name.outputs.string }}
+    steps:
+      # Uses the provided branch name or it defaults to the current
+      # branch, if empty, for the deployment slot name (legacy behaviour)
+      # Uses v6
+      - name: Get branch name
+        if: ${{ inputs.branch-name == '' }}
+        id: auto-branch-name
+        uses: tj-actions/branch-names@6871f53176ad61624f978536bbf089c574dc19a2
+
+      # Remove forward slash from name as its invalid for deployment slot name.
+      # Uses v2      
+      - name: Remove feature from branch name
+        id: rename-branch-name
+        uses: mad9000/actions-find-and-replace-string@27ff719b52870980e6ce19c288c5d3cc3ad202fa
```

Comment:
^ so we have pinned it to the given commit because a possibly *sketchy* source.  I believe that it should be mad9000/actions-find-and-replace-string@5 (and probably the SHA of the commit for that version, which is bce4d50390d08b67e1822bd72fb54b3cf11b0ce9).